### PR TITLE
Ipfs 0.4.16 release

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
cc @Stebalien @Mr0grog @schomatis

**Comment by Stebalien** [comment]
I think github screwed up somewhere.

**Comment by Stebalien** [comment]
Blocked on #5191.

**Comment by whyrusleeping** [comment]
oh boy

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+depend on, notably libp2p. This version of ipfs contains the changes for libp2p
+v6.0.5, In that release, we made significant changes to the codebase to allow
+for more easy integration of future transports and modules. We also improved
+some of our dialing logic, fixed a couple issues in the DHT, and added
+support for a new way to write ipfs multiaddrs.  The transport refactor now
+allows us to much more easily add in support for running libp2p (and by
+extension, ipfs) over QUIC, or using TLS instead of secio for encrypting
+connections. Our [QUIC
+transport](https://github.com/libp2p/go-libp2p-quic-transport) currently
+works, and can be plugged into libp2p manually (though note that it is
```
Comment (Mr0grog) [inline CHANGELOG.md]: Is there a doc, issue, or discuss post, or blog post we can point people to that describes how to add the QUIC transport manually?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,62 @@
 # go-ipfs changelog
 
+Ipfs 0.4.16 is a fairly small release in terms of changes to the ipfs codebase,
+but it contains a huge amount of changes and improvements from libraries we
+depend on, notably libp2p. This version of ipfs contains the changes for libp2p
+v6.0.5, In that release, we made significant changes to the codebase to allow
```
Comment (Mr0grog) [inline CHANGELOG.md]: Would it be helpful to say we upgraded from libp2p v5.0.14 to v6.0.5 so people know what range of changelog entries to look for if they want to dig into details on their own? (Also, do I have that version correct?)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,62 @@
...
+But caling them 'ipfs' multiaddrs is a bit misleading, as this is actually
+the multiaddr of a libp2p peer that runs ipfs. Other protocols using libp2p
+right now still have to use multiaddrs that say 'ipfs', without actually
+having anything to do with ipfs. Towards that, we are renaming them to
+'p2p' multiaddrs. Moving forward these addresses will be written as:
+`/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`.
+This release adds support for *parsing* both types of addresses into the
+same binary format, and the binary format is remaining exactly the same. A
+future release will have the ipfs daemon switch to *printing* out addresses
+this way, once a large enough portion of the network has upgraded.
```
Comment (Mr0grog) [inline CHANGELOG.md]: Whoa, this paragraph is huuuuge. Maybe a bulleted list would make it a bit easier to parse?

**Comment by Stebalien** [comment]
I've rewritten the release notes with paragraphs and sections. That required a lot of rewording (although everything should still be there) so someone should re-read them.

File: CHANGELOG.md (hunk 1 above)
Comment (Stebalien) [reply CHANGELOG.md]: No. Really, we don't want users to do this unless the know what they're doing (or ask us and we can talk it over with them).

File: CHANGELOG.md (hunk 2 above)
Comment (Stebalien) [reply CHANGELOG.md]: It says that further down but I've put that higher up in the version with paragraphs.

File: CHANGELOG.md (hunk 3 above)
Comment (Stebalien) [reply CHANGELOG.md]: Turned into paragraphs.

**Comment by Mr0grog** [comment]
Looks good to me 👍. I made some more very minor tweaks in #5198, but none of them are especially critical.

**Comment by Stebalien** [comment]
Blocked on https://github.com/ipfs/go-ipfs/issues/5197 and a bug I'm working through with @lgierth (possibly #5186?).

**Comment by whyrusleeping** [comment]
@Stebalien good to remove the blocked label?

**Pull request merged**